parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -783,7 +783,7 @@ public ParquetFileReader(InputFile file, ParquetReadOptions options, FileDecrypt
       f.close();
       throw e;
     }
-    if (null != fileDecryptionProperties && fileDecryptor.plaintextFile() && fileDecryptor.plaintextFilesAllowed()) {
+    if (null != fileDecryptionProperties && null != fileDecryptor && fileDecryptor.plaintextFile() && fileDecryptor.plaintextFilesAllowed()) {
       // Plaintext file. No need in decryptor

```

Comment:
How exceptions are possible here? Please see the lines 775-776. If fileDecryptionProperties is not null - then fileDecryptor  can not be null.

Reply:
Does the unit test pass on your side? I hit the exception when running TestEncryption() with NPE. The exception is as below. This change fixes it. 

java.lang.NullPointerException
	at org.apache.parquet.hadoop.ParquetFileReader.<init>(ParquetFileReader.java:823)
	at org.apache.parquet.hadoop.ParquetFileReader.open(ParquetFileReader.java:672)
	at org.apache.parquet.hadoop.ParquetReader.initReader(ParquetReader.java:161)
	at org.apache.parquet.hadoop.ParquetReader.read(ParquetReader.java:144)
	at org.apache.parquet.hadoop.TestEncryption.test(TestEncryption.java:170)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.rules.ExternalResource$1.evaluate(ExternalResource.java:48)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.execution.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:47)
	at com.intellij.rt.execution.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:242)
	at com.intellij.rt.execution.junit.JUnitStarter.main(JUnitStarter.java:70)

Reply:
Interesting.. I'll check this problem, thanks for the input.

Reply:
This test runs ok in my env. Given the lines numbers in the exception you get (823, 672 in ParquetFileReader), you probably use an old code version.

Reply:
I just checked my code. The exception is because there are some changes from our side which rename it. While merging your change, the exception happens. I will fix on my side. Yes, we do have the most recent code. I am glad the unit test catch that. 

Please go ahead to merge the first one and ignore this one. Thanks.

Reply:
Can you modify the pr, I don't see a button that allows to merge a subset of commits.